makeReducedConfig.py

```diff
@@ -0,0 +1,209 @@
+        if self.cinput.name != c_ref.cinput.name: return False
+        if self.coutput.name != c_ref.coutput.name: return False
+        if self.cinext != c_ref.cinext: return False
+        if self.coutext != c_ref.coutext: return False
+        return True
+
+class project:
+    def __init__(self):
+        self.connections = []
+        self.nodes = {}
+        self.tcs = []
+        self.l1phis = []
+        self.lxphis = []
+
+    def addNode(self, n):
+        if n.name not in self.nodes:
+            self.nodes[n.name] = n
+        else:
+            for c in n.connections:
+                self.nodes[n.name].addConnection(c)
+
+    def addConnection(self, c):
+        for c_ref in self.connections:
+            if c.isSame(c_ref): return
+        self.connections.append(c)
+
+    # Load in a project from a wires file
+    def loadProject(self, fname):
+        with open(fname, "r") as f:
+            for line in f:
+
+                # Get raw names from wires file
+                mem = line.split("input=>")[0].strip()
+                str_in = line.split("input=>")[1].split("output=>")[0].strip()
+                str_out = line.split("output=>")[1].strip()
+
+                # Create nodes and extensions from strings
+                node_in = node(str_in.split(".")[0])
+                in_ext = str_in.split(".")[1] if "." in str_in else ""
+                node_out = node(str_out.split(".")[0])
+                out_ext = str_out.split(".")[1] if "." in str_out else ""
+
+                # Add connection and nodes to project
+                new_c = connection(mem, node_in, node_out, in_ext, out_ext)
+                self.addConnection(new_c)
+                node_in.addConnection(new_c)
+                node_out.addConnection(new_c)
+                self.addNode(node_in)
+                self.addNode(node_out)
+
+    # Save a project to a wires file
+    def saveProject(self, fname):
+        with open(fname, "w") as f:
+            for c in self.connections:
+                f.write(c.printConnection()+"\n")
+
+    def addTC(self, tc, ref_p):
+        tcs = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L"]
+        l1phis = ["A", "B", "C", "D", "E", "F", "G", "H"]
+        lxphis = ["A", "B", "C", "D"]
+
+        if tc.upper() not in tcs:
+            print "Bad TC index, not adding."
+            return
+
+        tc_i = tcs.index(tc.upper())
+        phi1_i = int(tc_i*1.*len(l1phis)/len(tcs))
+        phix_i = int(tc_i*1.*len(lxphis)/len(tcs))
+
+        print "Adding regions to project: TC_L1l2%s, L1PHI%s, LxPHI%s"%(tcs[tc_i], l1phis[phi1_i], lxphis[phix_i])
+        self.tcs.append(tcs[tc_i])
+        self.l1phis.append(l1phis[phi1_i])
+        self.lxphis.append(lxphis[phix_i])
+
+        # Add nodes and connections to the project
+        # Starting with TC_L1L2x and moving up and down the chain
+
+        print "Starting with node TC_L1L2%s"%tcs[tc_i]
+        n = node("TC_L1L2%s"%tcs[tc_i])
```

Comment:
Why does the script start at TC, rather than any other processing module in the chain? And could the layer pair L1L2 be made configurable via an argument passed to the script (with default value "L1L2")?

Reply:
Sorry -- didn't have my email notifications set apparently, and just seeing these now! I'll implement all your other comments. 

For this one, I picked starting with the TC because it has the finest segmentation in phi, so if you pick a given TC it's unambiguous which phi segmentation you want in L1 and in the rest of the layers. Nonetheless, the algorithm I created is flexible and will do something reasonable no matter where you start, this just gives you the most precise control. 

There are a few places you could start that would cause trouble, basically if you were in the middle of some connection loop. The algorithm recursively looks for inputs of inputs and outputs of outputs to the original node. It won't look for outputs of an input (to prevent infinite recursion). So for example if you started with an ME, it would find that a PR was its input and an MC was it's output, but not that the PR outputted to the MC. 

Both these things could be modified if needed, but if not, I think they make for a clean implementation.

Making the L1L2 is trivial of course, I can add that.

Reply:
Done adding configurable L1L2